### PR TITLE
[3.7] bpo-35283: Add a deprecated warning for the threading.Thread.isAlive

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue35283

### CODE REVIEW

**Comment by corona10** [comment]
@asvetlov @vstinner 
Please take a look, this PR is a backport version of #11454 
Thanks!

File: Lib/threading.py
Comment (vstinner) [inline Lib/threading.py]: I suggest to remove the class name: just write is_alive().

File: Lib/threading.py
Comment (vstinner) [inline Lib/threading.py]: Same here, maybe omit the class name.

File: Lib/threading.py
Comment (vstinner) [inline Lib/threading.py]: No, please don't add new methods to stable branches (3.7). Revert this change from your PR.

File: Lib/threading.py
Comment (vstinner) [inline Lib/threading.py]: I would prefer to avoid always importing warnings, since up to date code is not supposed to emit a warning. I suggest to move the import inside the function calls (in each isAlive() function).

**Review by vstinner — CHANGES_REQUESTED** [review]

**Comment by vstinner** [comment]
In practice, I would prefer to start with a change in the master branch, and later consider what to do in the 3.7 branch. I'm not sure if it's ok to start to emit a PendingDeprecatinonWarning in 3.7.3.

**Comment by corona10** [comment]
I have made the requested changes; please review again.
Also, I reflect your review on the master branch PR #11454 
thanks!

**Comment by eamanu** [comment]
> I have made the requested changes; please review again.
> Also, I reflect your review on the master branch PR #11454
> thanks!

IMO you have to close this PR, and then when the change is accepted on master branch (#11454), will be backport to 3.x

**Comment by corona10** [comment]
@eamanu 

Most of the case, the backport patch you mentioned can be written by a bot.
(e.g https://github.com/python/cpython/pull/7204) So, we don't have to open seperate PR.

But in this case, as vstiner mentioned, we should apply `PendingDeprecationWarning` for python 3.7.
So for to do this, IMHO manual backport patch is a right way. if not
python3.7 will have a weird change to `isAlive()` method with deprecating message. 
`DeprecationWarning` -> `PendingDeprecationWarning`

So I think that this PR can be merged after #11454 is merged when core developers decide to emit `PendingDeprecationWarning` to 3.7 also .

**Comment by eamanu** [comment]
> @eamanu
> 
> Most of the case, the backport patch you mentioned can be written by a bot.
> (e.g #7204) So, we don't have to open seperate PR.
> 
> But in this case, as vstiner mentioned, we should apply `PendingDeprecationWarning` for python 3.7.
> So for to do this, IMHO manual backport patch is a right way. if not
> python3.7 will have a weird change to `isAlive()` method with deprecating message.
> `DeprecationWarning` -> `PendingDeprecationWarning`
> 
> So I think that this PR can be merged after #11454 is merged when core developers decide to emit `PendingDeprecationWarning` to 3.7 also .

Yes you right. But the when the bot generate the backport, maintainer could modify the PR, isn't?

**Comment by corona10** [comment]
@eamanu 
Yeah.. but that will also need manually update the commit.
So sometimes in my experience, I wrote a manual backport patch for other python versions.
ex) https://github.com/python/cpython/pull/8156

So do I have to close this PR?

**Pull request closed without merging**
